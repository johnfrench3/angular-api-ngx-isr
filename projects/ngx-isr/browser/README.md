# ngx-isr/browser

Secondary entry point of `ngx-isr`. It can be used by importing from `ngx-isr/browser`.
