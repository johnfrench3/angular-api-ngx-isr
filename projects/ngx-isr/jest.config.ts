/* eslint-disable */
export default {
  displayName: 'ngx-isr',
  preset: '../../jest.preset.js',
  coverageDirectory: '../../coverage/projects/ngx-isr',
};
