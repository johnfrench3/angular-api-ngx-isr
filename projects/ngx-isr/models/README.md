# ngx-isr/models

Secondary entry point of `ngx-isr`. It can be used by importing from `ngx-isr/models`.
