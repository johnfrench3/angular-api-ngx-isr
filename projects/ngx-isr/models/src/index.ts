export {
  CacheHandler,
  CacheISRConfig,
  CacheISRConfig as ISROptions,
  CacheData,
} from './cache-handler';

export {
  ISRHandlerConfig,
  InvalidateConfig,
  RenderConfig,
  ServeFromCacheConfig,
  RouteISRConfig,
} from './isr-handler-config';

export { INgxIsrService, NgxIsrState } from './ngx-isr-service.interface';
