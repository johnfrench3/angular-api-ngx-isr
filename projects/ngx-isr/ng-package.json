{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../dist/ngx-isr",
  "lib": {
    "entryFile": "src/public-api.ts"
  },
  "allowedNonPeerDependencies": ["ng-morph"]
}
