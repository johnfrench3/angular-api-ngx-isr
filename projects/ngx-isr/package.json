{
  "name": "ngx-isr",
  "author": "Enea Jahollari",
  "description": "Incremental Static Regeneration for Angular",
  "version": "0.6.0",
  "engines": {
    "node": "^16.13.0 || ^18.10.0",
    "npm": "^9.5.0",
    "yarn": "^1.22.0"
  },
  "peerDependencies": {
    "@angular/common": ">=15.0.0 || >=16.0.0",
    "@angular/core": ">=15.0.0 || >=16.0.0",
    "@types/node": "^14.15.0 || ^16.0.0 || ^18.0.0"
  },
  "dependencies": {
    "tslib": "^2.3.0",
    "ng-morph": "^2.1.3"
  },
  "keywords": [
    "Angular",
    "Library",
    "TypeScript",
    "Angular Universal",
    "Incremental Static Regeneration",
    "Angular ISR"
  ],
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/eneajaho/ngx-isr"
  },
  "schematics": "./schematics/collection.json",
  "ng-add": {
    "save": "dependencies"
  },
  "ng-update": {
    "migrations": "./schematics/migration.json"
  }
}
