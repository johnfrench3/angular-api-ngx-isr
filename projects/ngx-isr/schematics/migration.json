{
  "$schema": "../../../node_modules/@angular-devkit/schematics/collection-schema.json",
  "schematics": {
    "migrate-to-rx-angular-isr": {
      "version": "0.6.0",
      "description": "Migrate to @rx-angular/isr and new class names",
      "factory": "./src/migrations/migrate-to-rx-angular-isr/index"
    }
  }
}
