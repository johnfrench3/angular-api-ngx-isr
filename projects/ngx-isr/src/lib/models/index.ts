export * from './cache-handler';
export * from './isr-handler-config';