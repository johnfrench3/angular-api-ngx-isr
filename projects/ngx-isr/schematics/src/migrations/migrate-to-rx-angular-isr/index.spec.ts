import {Tree} from '@angular-devkit/schematics';
import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import * as path from 'path';

describe('isr migrate-to-rx-angular-isr', () => {
  let appTree: UnitTestTree | undefined;

  it('should replace ngx-isr/server and class names', async () => {
    appTree = await setupTestFile(`
      import { BrowserModule } from '@angular/platform-browser';
      import { NgxIsrModule,
               provideISR,
               ISRHandler,
               FileSystemCacheHandler,
               FileSystemCacheOptions
             } from 'ngx-isr/server';
      import { AppComponent } from './app.component';

      @NgModule({
        declarations: [
          AppComponent,
        ],
        imports: [
          BrowserModule,
          NgxIsrModule.forRoot()
        ],
        bootstrap: [AppComponent]
      })
      export class AppModule { }
  `);

    const file = appTree?.readContent('app.module.ts');

    // @ts-ignore
    expect(file).toMatchSnapshot();
  });

  it('should replace ngx-isr/models and class names', async () => {
    appTree = await setupTestFile(`
      import { BrowserModule } from '@angular/platform-browser';
      import { INgxIsrService, NgxIsrState } from 'ngx-isr/models';
      import { AppComponent } from './app.component';

      @NgModule({
        declarations: [
          AppComponent,
        ],
        imports: [
          BrowserModule
        ],
        bootstrap: [AppComponent]
      })
      export class AppModule { }
  `);

    const file = appTree?.readContent('app.module.ts');

    // @ts-ignore
    expect(file).toMatchSnapshot();
  });

  it('should replace ngx-isr/browser and class names', async () => {
    appTree = await setupTestFile(`
      import { BrowserModule } from '@angular/platform-browser';
      import { NgxIsrService } from 'ngx-isr/browser';
      import { AppComponent } from './app.component';

      @NgModule({
        declarations: [
          AppComponent,
        ],
        imports: [
          BrowserModule
        ],
        providers: [NgxIsrService],
        bootstrap: [AppComponent]
      })
      export class AppModule { }
  `);

    const file = appTree?.readContent('app.module.ts');

    // @ts-ignore
    expect(file).toMatchSnapshot();
  });

  function setupTestFile(fileInput: string, filePath = './app.module.ts') {
    const runner = new SchematicTestRunner(
      'ngx-isr',
      path.join(__dirname, '../../../migration.json')
    );
    const tree = new UnitTestTree(Tree.empty());

    tree.create(filePath, fileInput);

    return runner.runSchematic(`migrate-to-rx-angular-isr`, {}, tree);
  }

});
