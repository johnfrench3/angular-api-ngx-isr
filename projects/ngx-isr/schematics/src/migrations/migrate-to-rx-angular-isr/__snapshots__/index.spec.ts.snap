// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isr migration update-16.0.0 should replace ngx-isr/browser 1`] = `
"import { BrowserModule } from '@angular/platform-browser';
import { NgxIsrService } from 'ngx-isr/browser';
import { AppComponent } from './app.component';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule],
  providers: [NgxIsrService],
  bootstrap: [AppComponent],
})
export class AppModule {}
"
`;

exports[`isr migration update-16.0.0 should replace ngx-isr/browser and class names 1`] = `
"import { IsrService } from '@rx-angular/isr/browser';

import { BrowserModule } from '@angular/platform-browser';
import { AppComponent } from './app.component';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule],
  providers: [IsrService],
  bootstrap: [AppComponent],
})
export class AppModule {}
"
`;

exports[`isr migration update-16.0.0 should replace ngx-isr/models 1`] = `
"import { BrowserModule } from '@angular/platform-browser';
import { INgxIsrService, NgxIsrState } from 'ngx-isr/models';
import { AppComponent } from './app.component';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule],
  bootstrap: [AppComponent],
})
export class AppModule {}
"
`;

exports[`isr migration update-16.0.0 should replace ngx-isr/models and class names 1`] = `
"import { IsrServiceInterface, IsrState } from '@rx-angular/isr/models';

import { BrowserModule } from '@angular/platform-browser';
import { AppComponent } from './app.component';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule],
  bootstrap: [AppComponent],
})
export class AppModule {}
"
`;

exports[`isr migration update-16.0.0 should replace ngx-isr/server 1`] = `
"import {
  IsrModule,
  provideISR,
  ISRHandler,
  FileSystemCacheHandler,
  FileSystemCacheOptions,
} from '@rx-angular/isr/server';

import { BrowserModule } from '@angular/platform-browser';
import { AppComponent } from './app.component';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule, IsrModule.forRoot()],
  bootstrap: [AppComponent],
})
export class AppModule {}
"
`;

exports[`isr migration update-16.0.0 should replace ngx-isr/server and class names 1`] = `
"import {
  IsrModule,
  provideISR,
  ISRHandler,
  FileSystemCacheHandler,
  FileSystemCacheOptions,
} from '@rx-angular/isr/server';

import { BrowserModule } from '@angular/platform-browser';
import { AppComponent } from './app.component';

@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule, IsrModule.forRoot()],
  bootstrap: [AppComponent],
})
export class AppModule {}
"
`;
